metricbeat/module/system/raid/raid.go

```diff
@@ -0,0 +1,76 @@
+// +build linux
+
+package raid
+
+import (
+	"path/filepath"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/common/cfgwarn"
+	"github.com/elastic/beats/metricbeat/mb"
+	"github.com/elastic/beats/metricbeat/mb/parse"
+	"github.com/elastic/beats/metricbeat/module/system"
+	"github.com/elastic/procfs"
+
+	"github.com/pkg/errors"
+)
+
+func init() {
+	if err := mb.Registry.AddMetricSet("system", "raid", New, parse.EmptyHostParser); err != nil {
+		panic(err)
+	}
+}
+
+type MetricSet struct {
+	mb.BaseMetricSet
+	fs procfs.FS
+}
+
+func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
+	cfgwarn.Experimental("The system raid metricset is experimental")
+
+	systemModule, ok := base.Module().(*system.Module)
+	if !ok {
+		return nil, errors.New("unexpected module type")
+	}
+
+	mountPoint := filepath.Join(systemModule.HostFS, procfs.DefaultMountPoint)
+	fs, err := procfs.NewFS(mountPoint)
```

Comment:
For testing purposes I'd make it possible to fully define the proc dir location so that you can create `testdata/proc/mdstat` and read it.

Reply:
Here's some sample data for testing: https://github.com/elastic/procfs/blob/master/fixtures/mdstat